### PR TITLE
Homework (Branch Random)

### PR DESCRIPTION
**Pull request by Leah2356**

### CODE REVIEW

File: src/main/java/feiner/fallingsand/Sand.java [hunk 1]
```diff
@@ -34,15 +45,42 @@ public void put(int x, int y) {
...
-}
+
+    /**
+     * Add a method called randomSand(int n).
+     * This will add n pieces of sand into the field in random positions
+     */
+    public void randomSand(int n) {
+        for (int i = 0; i < n; i++) {
+            int y = random.nextInt(field.length);
+            int x = random.nextInt(field.length);
```
Comment (andrewoid) [inline src/main/java/feiner/fallingsand/Sand.java]: There's a bug here

File: src/main/java/feiner/fallingsand/Sand.java [hunk 2]
```diff
@@ -34,15 +45,42 @@ public void put(int x, int y) {
...
+                    boolean rightFirst = random.nextBoolean();
+                    int direction1 = rightFirst ? 1 : -1;
+                    int direction2 = rightFirst ? -1 : 1;
+                    if (field[y + 1][x + direction1] == 0) {
+                        field[y][x] = 0;
+                        field[y + 1][x + direction1] = 1;
+                    } else if (field[y + 1][x + direction2] == 0) {
+                        field[y][x] = 0;
+                        field[y + 1][x + direction2] = 1;
+                    } else {
```
Comment (andrewoid) [inline src/main/java/feiner/fallingsand/Sand.java]: What does this line do?

**Comment by Leah2356** [comment]
Hi, I see that you commented asking what the line does. We did this in
class in order to determine if the sand should fall to the right or to the
left.

File: src/main/java/feiner/fallingsand/Sand.java (hunk 2 above)
Comment (andrewoid) [reply src/main/java/feiner/fallingsand/Sand.java]: Specifically talking about line 68

Comment (Leah2356) [reply src/main/java/feiner/fallingsand/Sand.java]: Oh, I was not sure which line you were referring to. The else statement is there to handle any instances where the if and the else if is not met. I thought we should add that line to handle it to show that the space of (y,x) is already occupied.

Comment (andrewoid) [reply src/main/java/feiner/fallingsand/Sand.java]: But we know that the space is already occupied.
Does line 68 do anything?

Comment (Leah2356) [reply src/main/java/feiner/fallingsand/Sand.java]: No it doesn't really do anything, should I delete it?

Comment (andrewoid) [reply src/main/java/feiner/fallingsand/Sand.java]: What do you think?

Comment (Leah2356) [reply src/main/java/feiner/fallingsand/Sand.java]: Yes, I deleted it.

**Review by andrewoid — APPROVED** [review]

**Pull request merged**